msbuild/tests/Xamarin.iOS.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -0,0 +1,99 @@
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Utilities;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Collections.Generic;
+
+namespace Xamarin.iOS.Tasks {
+	[TestFixture]
+	public class LocalizationStringTest : TestBase {
+
+		[Test]
+		public void OperatingSystemTranslations ()
+		{
+			CultureInfo culture1 = CultureInfo.CurrentCulture;
+			string locale = culture1.Name;
+
+			var task = CreateTask<CollectITunesArtwork> ();
+			task.ITunesArtwork = new TaskItem [] { new TaskItem (Assembly.GetExecutingAssembly ().Location) };
+
+			Assert.IsFalse (task.Execute (), "Execute failure");
+			Assert.AreEqual (1, Engine.Logger.ErrorEvents.Count, "ErrorCount");
+			bool isTranslated = TranslationAvailable (locale, Engine.Logger.ErrorEvents [0].Message);
+			Assert.IsTrue (isTranslated, "Your current locale is not supported: " + locale + ".");
+		}
+
+		[Test]
+		public void AllSupportedTranslations ()
```

Comment:
It's not a big deal, but this can be simplified somewhat by using the `[TestCase]` attribute:

```csharp
[Test]
[TestCase ("cs-CZ")]
[TestCase ("de-DE")]
[TestCase ("en-US")]
...
[TestCase ("zh-TW")]
public void SupportedTranslations (string culture)
{
    var newCulture = new CultureInfo (culture);
    
}
```

Also, this way you don't have to keep track of the failures and assert at the end, each culture will be tested separately.

Reply:
I think we can even do better, TestCase could take the local AND the expected message, we can remove public bool TranslationAvailable (string locale, string respectiveErrorMessage) out since would be a two liner, and move to:
```csharp
[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
public void SupportedTranslations (string culture, string expectedMessage)
{
...
}
```
[Test] is not needed if you are using [TestCase].

Reply:
This made it a lot cleaner!